File: session-02-dependency-injection/python/menu/repository.py
from abc import ABC, abstractmethod

from menu.model import MenuItem


class MenuRepository(ABC):

    @abstractmethod
    def find_all(self) -> list[MenuItem]:
        ...


class InMemoryMenuRepository(MenuRepository):

    def __init__(self):
        self._items = [
            MenuItem(id=1, name="Latte", price=4.50),
            MenuItem(id=2, name="Espresso", price=3.00),
            MenuItem(id=3, name="Cappuccino", price=4.00),
            MenuItem(id=4, name="Americano", price=3.50),
        ]

    def find_all(self) -> list[MenuItem]:
        return self._items

File: session-03-who-owns-the-main-loop/go/menu/repository.go
package menu

type MenuRepository interface {
	FindAll() []MenuItem
}

type InMemoryMenuRepository struct {
	items []MenuItem
}

func NewInMemoryMenuRepository() *InMemoryMenuRepository {
	return &InMemoryMenuRepository{
		items: []MenuItem{
			{ID: 1, Name: "Latte", Price: 4.50},
			{ID: 2, Name: "Espresso", Price: 3.00},
			{ID: 3, Name: "Cappuccino", Price: 4.00},
			{ID: 4, Name: "Americano", Price: 3.50},
		},
	}
}

func (r *InMemoryMenuRepository) FindAll() []MenuItem {
	return r.items
}
